// @sap-cloud-sdk/generator-common/dist/sdk-metadata/sdk-metadata.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceStatusValues = exports.sdkMetadataHeaderIntroText = exports.getLevenshteinClosest = exports.getSdkMetadataClient = exports.getVersionForClient = exports.sdkMetadataHeader = exports.getSdkMetadataFileNames = void 0;
const util_1 = require("@sap-cloud-sdk/util");
const fast_levenshtein_1 = __importDefault(require("fast-levenshtein"));
const util_2 = require("./util");
const distanceThreshold = 5;
/**
 * @internal
 */
function getSdkMetadataFileNames(originalFileName) {
    return {
        clientFileName: `${originalFileName}_CLIENT_JS.json`,
        headerFileName: `${originalFileName}_HEADER.json`
    };
}
exports.getSdkMetadataFileNames = getSdkMetadataFileNames;
/**
 * @internal
 */
async function sdkMetadataHeader(type, originalFileName, versionInPackageJson) {
    return {
        type,
        // For the file name with use the artifact.name from API which should be the unique identifier
        name: (0, util_1.removeFileExtension)(originalFileName),
        version: await getVersionForClient(versionInPackageJson),
        introText: exports.sdkMetadataHeaderIntroText
    };
}
exports.sdkMetadataHeader = sdkMetadataHeader;
/**
 * @internal
 */
async function getVersionForClient(versionInPackageJson) {
    return versionInPackageJson || (0, util_2.getSdkVersion)();
}
exports.getVersionForClient = getVersionForClient;
/**
 * @internal
 */
function getSdkMetadataClient(generationAndUsage, pregeneratedLibrary) {
    const status = pregeneratedLibrary
        ? exports.ServiceStatusValues.certified
        : exports.ServiceStatusValues.verified;
    return {
        language: 'JavaScript',
        serviceStatus: status,
        pregeneratedLibrary,
        generationAndUsage
    };
}
exports.getSdkMetadataClient = getSdkMetadataClient;
/**
 * Gets the closest matching object using Levenshtein distance algorithm.
 * @param name - Name of the service or api class.
 * @param objectsToCheck - List of objects, e.g. VdmEntity, FunctionImports, etc.
 * @param extractorFn - Function to get the object's property to match against name.
 * @returns - closest matched object or undefined if not found.
 * @internal
 */
function getLevenshteinClosest(name, objectsToCheck, extractorFn) {
    const distBelowThreshold = objectsToCheck.reduce((prev, obj) => {
        const levenshteinDist = getLevenshteinDistance(name, extractorFn(obj));
        if (levenshteinDist < distanceThreshold) {
            return [...prev, { dist: levenshteinDist, obj }];
        }
        return prev;
    }, []);
    if (distBelowThreshold.length > 0) {
        return distBelowThreshold.sort((a, b) => (a.dist < b.dist ? -1 : 1))[0].obj;
    }
}
exports.getLevenshteinClosest = getLevenshteinClosest;
function getLevenshteinDistance(stringA, stringB) {
    return fast_levenshtein_1.default.get(getSanitizedString(stringA), getSanitizedString(stringB));
}
function getSanitizedString(text) {
    return text.replace(/[^A-Za-z]/g, '').toLowerCase();
}
/**
 * @internal
 */
exports.sdkMetadataHeaderIntroText = 'The SAP Cloud SDK is a versatile set of libraries and tools for developers to build cloud-native applications on the SAP Business Technology Platform. To simplify the consumption of multiple services published on the SAP API Business Hub, the SDK offers code generators for OData and OpenAPI together with pregenerated typed client libraries. These libraries are seamlessly integrated with connectivity, authentication, multi-tenancy, and other abstractions to speed up application development.';
/**
 * @internal
 */
exports.ServiceStatusValues = {
    certified: {
        status: 'certified',
        statusText: 'API Client available for download',
        statusLongText: 'You can download our pregenerated API client for this API.'
    },
    verified: {
        status: 'verified',
        statusText: 'API Client generation tested but no download available',
        statusLongText: 'The SAP Cloud SDK team checked for this API that generating an API client works, but no pregenerated client is available. You can generate your own client for this API.'
    },
    unknown: {
        status: 'unknown',
        statusText: 'API Client generation not tested',
        statusLongText: 'The SAP Cloud SDK team has not tested if generating an API client for this API works. You might try to generate a client for this API if you need one.'
    },
    failed: {
        status: 'failed',
        statusText: 'API Client generation has failed',
        statusLongText: 'The SAP Cloud SDK team checked for this API if generating an API client works, which has failed. You might try it on your own.'
    }
};
