// @sap-cloud-sdk/generator/dist/file-generator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.copyFile = exports.otherFile = exports.sourceFile = void 0;
const fs_1 = require("fs");
const path_1 = require("path");
const util_1 = require("@sap-cloud-sdk/util");
const internal_1 = require("@sap-cloud-sdk/generator-common/internal");
const logger = (0, util_1.createLogger)({
    package: 'generator',
    messageContext: 'file-generator'
});
/**
 * @internal
 */
function sourceFile(directory, relativePath, content, overwrite) {
    const file = directory.createSourceFile(`${relativePath}.ts`, addFileComment(content), {
        overwrite
    });
    file.formatText({ insertSpaceAfterCommaDelimiter: true });
    return file;
}
exports.sourceFile = sourceFile;
/**
 * @internal
 */
function otherFile(directory, relativePath, content, overwrite) {
    return directory.createSourceFile(relativePath, content, {
        overwrite
    });
}
exports.otherFile = otherFile;
/**
 * @internal
 */
function copyFile(fromPath, toRelativePath, toDirectory, overwrite) {
    try {
        const fileContent = (0, fs_1.readFileSync)((0, path_1.resolve)(fromPath), { encoding: 'utf8' });
        toDirectory.createSourceFile(toRelativePath, fileContent, {
            overwrite
        });
    }
    catch (err) {
        logger.error(err);
    }
}
exports.copyFile = copyFile;
function addFileComment(content) {
    content.leadingTrivia = (0, internal_1.getCopyrightHeader)();
    return content;
}
