// @sap-cloud-sdk/generator/dist/sdk-metadata/generation-and-usage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getODataLinks = exports.getApiSpecificUsage = exports.getGenericGenerationAndUsage = exports.linkGenerationDocumentation = exports.getGenerationAndUsage = void 0;
const internal_1 = require("@sap-cloud-sdk/generator-common/internal");
const code_samples_1 = require("./code-samples");
const code_sample_util_1 = require("./code-sample-util");
/**
 * @internal
 */
async function getGenerationAndUsage(service) {
    return {
        ...(await getGenericGenerationAndUsage()),
        apiSpecificUsage: getApiSpecificUsage(service)
    };
}
exports.getGenerationAndUsage = getGenerationAndUsage;
/**
 * @internal
 */
exports.linkGenerationDocumentation = 'https://sap.github.io/cloud-sdk/docs/js/features/odata/generate-odata-client';
// will be used to generate metadata for failed and unknown case.
/**
 * @internal
 */
async function getGenericGenerationAndUsage() {
    return {
        genericUsage: (0, code_samples_1.genericEntityCodeSample)(),
        repository: 'npm',
        apiSpecificUsage: undefined,
        links: getODataLinks(),
        generationSteps: (0, internal_1.getGenerationSteps)('npm install -g @sap-cloud-sdk/generator', 'generate-odata-client --inputDir path/to/specification/ --outputDir path/to/client/', exports.linkGenerationDocumentation, 'OData'),
        generatorVersion: await (0, internal_1.getSdkVersion)(),
        generatorRepositoryLink: 'https://www.npmjs.com/package/@sap-cloud-sdk/generator'
    };
}
exports.getGenericGenerationAndUsage = getGenericGenerationAndUsage;
/**
 * @internal
 */
function getApiSpecificUsage(service) {
    var _a, _b;
    if (((_a = service.entities) === null || _a === void 0 ? void 0 : _a.length) > 0) {
        const entity = (0, code_sample_util_1.getODataEntity)(service.originalFileName, service.entities);
        return {
            ...(0, code_samples_1.entityCodeSample)(entity.className, service.className, service.npmPackageName),
            header: internal_1.usageHeaderText
        };
    }
    // Return function/action import usage if no entity is found.
    if (((_b = service.functionImports) === null || _b === void 0 ? void 0 : _b.length) > 0) {
        const functionImport = (0, code_sample_util_1.getActionFunctionImport)(service.originalFileName, service.functionImports);
        return {
            ...(0, code_samples_1.functionImportCodeSample)(functionImport, `${service.npmPackageName}/function-imports`),
            header: internal_1.usageHeaderText
        };
    }
    if (service.actionImports) {
        const actionImport = (0, code_sample_util_1.getActionFunctionImport)(service.originalFileName, service.actionImports);
        return {
            ...(0, code_samples_1.actionImportCodeSample)(actionImport, `${service.npmPackageName}/action-imports`),
            header: internal_1.usageHeaderText
        };
    }
    return {
        instructions: '',
        text: '',
        header: internal_1.usageHeaderText
    };
}
exports.getApiSpecificUsage = getApiSpecificUsage;
/**
 * @internal
 */
function getODataLinks() {
    return (0, internal_1.getLinks)('https://sap.github.io/cloud-sdk/docs/js/features/odata/execute-odata-request', exports.linkGenerationDocumentation, 'OData');
}
exports.getODataLinks = getODataLinks;
