// @sap-cloud-sdk/generator/dist/edmx-to-vdm/common/action-function-parameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getFunctionImportParameters = exports.getActionImportParameters = void 0;
const generator_utils_1 = require("../../generator-utils");
const description_util_1 = require("../description-util");
const edmx_to_vdm_util_1 = require("../edmx-to-vdm-util");
/**
 * @internal
 */
function getActionImportParameters(edmxActionImport, edmxParameters, swaggerDefinition, formatter) {
    return getParameter(edmxActionImport, edmxParameters, swaggerDefinition, formatter);
}
exports.getActionImportParameters = getActionImportParameters;
/**
 * @internal
 */
function getFunctionImportParameters(edmxFunctionImport, edmxParameters, swaggerDefinition, formatter) {
    return getParameter(edmxFunctionImport, edmxParameters, swaggerDefinition, formatter);
}
exports.getFunctionImportParameters = getFunctionImportParameters;
function getParameter(edmxActionFunctionImport, edmxParameters, swaggerDefinition, formatter) {
    return edmxParameters.map(p => {
        const swaggerParameter = swaggerDefinition
            ? swaggerDefinition.parameters.find(param => param.name === p.Name)
            : undefined;
        const typeMapping = (0, edmx_to_vdm_util_1.getTypeMappingActionFunction)(p.Type);
        return {
            originalName: p.Name,
            parameterName: formatter.originalToParameterName(edmxActionFunctionImport.Name, p.Name),
            edmType: typeMapping.edmType,
            jsType: typeMapping.jsType,
            fieldType: typeMapping.fieldType,
            nullable: (0, generator_utils_1.isNullableProperty)(p),
            description: (0, description_util_1.parameterDescription)(p, swaggerParameter)
        };
    });
}
