// @sap-cloud-sdk/generator/dist/action-function-import/export-statement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.exportStatement = void 0;
const util_1 = require("@sap-cloud-sdk/util");
const ts_morph_1 = require("ts-morph");
/**
 * @internal
 */
function exportStatement(actionFunctionImports, name) {
    return {
        kind: ts_morph_1.StructureKind.VariableStatement,
        declarationKind: ts_morph_1.VariableDeclarationKind.Const,
        declarations: [
            {
                name,
                initializer: exportsInitializer(actionFunctionImports)
            }
        ],
        isExported: true
    };
}
exports.exportStatement = exportStatement;
function exportsInitializer(actionFunctionImports) {
    return (actionFunctionImports.reduce((initializer, currentImport) => {
        if (initializer !== `{${util_1.unixEOL}`) {
            initializer += `,${util_1.unixEOL}`;
        }
        initializer += `${currentImport.name}`;
        return initializer;
    }, `{${util_1.unixEOL}`) + `${util_1.unixEOL}}`);
}
