// @sap-cloud-sdk/generator/dist/action-function-import/file.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.functionImportSourceFile = exports.actionImportSourceFile = void 0;
const util_1 = require("@sap-cloud-sdk/util");
const ts_morph_1 = require("ts-morph");
const parameters_interface_1 = require("./parameters-interface");
const export_statement_1 = require("./export-statement");
const action_1 = require("./action");
const import_1 = require("./import");
const function_1 = require("./function");
/**
 * @internal
 */
function actionImportSourceFile(service) {
    if (!service.actionImports) {
        throw new Error('Tried to create action import source files without actions in service metadata.');
    }
    return {
        kind: ts_morph_1.StructureKind.SourceFile,
        statements: [
            ...(0, import_1.importDeclarationsFunction)(service),
            ...(0, util_1.flat)(service.actionImports.map(action => actionImportStatements(action, service))),
            (0, export_statement_1.exportStatement)(service.actionImports, 'actionImports')
        ]
    };
}
exports.actionImportSourceFile = actionImportSourceFile;
function actionImportStatements(actionImport, service) {
    return [
        (0, parameters_interface_1.parametersInterface)(actionImport),
        (0, action_1.actionImportFunction)(actionImport, service)
    ];
}
/**
 * @internal
 */
function functionImportSourceFile(service) {
    return {
        kind: ts_morph_1.StructureKind.SourceFile,
        statements: [
            ...(0, import_1.importDeclarationsAction)(service),
            ...(0, util_1.flat)(service.functionImports.map(fi => functionImportStatements(fi, service))),
            (0, export_statement_1.exportStatement)(service.functionImports, 'functionImports')
        ]
    };
}
exports.functionImportSourceFile = functionImportSourceFile;
function functionImportStatements(functionImport, service) {
    return [
        (0, parameters_interface_1.parametersInterface)(functionImport),
        (0, function_1.functionImportFunction)(functionImport, service)
    ];
}
