// @sap/cloud-sdk-vdm-sales-order-service/SalesOrderItmPrecdgProcFlowApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesOrderItmPrecdgProcFlowApi = void 0;
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This is a generated file powered by the SAP Cloud SDK for JavaScript.
 */
const SalesOrderItmPrecdgProcFlow_1 = require("./SalesOrderItmPrecdgProcFlow");
const SalesOrderItmPrecdgProcFlowRequestBuilder_1 = require("./SalesOrderItmPrecdgProcFlowRequestBuilder");
const odata_v2_1 = require("@sap-cloud-sdk/odata-v2");
class SalesOrderItmPrecdgProcFlowApi {
    constructor(deSerializers = odata_v2_1.defaultDeSerializers) {
        this.entityConstructor = SalesOrderItmPrecdgProcFlow_1.SalesOrderItmPrecdgProcFlow;
        this.deSerializers = deSerializers;
    }
    _addNavigationProperties(linkedApis) {
        this.navigationPropertyFields = {
            TO_SALES_ORDER: new odata_v2_1.OneToOneLink('to_SalesOrder', this, linkedApis[0]),
            TO_SALES_ORDER_ITEM: new odata_v2_1.OneToOneLink('to_SalesOrderItem', this, linkedApis[1])
        };
        return this;
    }
    requestBuilder() {
        return new SalesOrderItmPrecdgProcFlowRequestBuilder_1.SalesOrderItmPrecdgProcFlowRequestBuilder(this);
    }
    entityBuilder() {
        return (0, odata_v2_1.entityBuilder)(this);
    }
    customField(fieldName, isNullable = false) {
        return new odata_v2_1.CustomField(fieldName, this.entityConstructor, this.deSerializers, isNullable);
    }
    get schema() {
        const fieldBuilder = new odata_v2_1.FieldBuilder(SalesOrderItmPrecdgProcFlow_1.SalesOrderItmPrecdgProcFlow, this.deSerializers);
        return {
            /**
         * Static representation of the [[salesOrder]] property for query construction.
         * Use to reference this property in query operations such as 'select' in the fluent request API.
         */
            SALES_ORDER: fieldBuilder.buildEdmTypeField('SalesOrder', 'Edm.String', false),
            /**
             * Static representation of the [[salesOrderItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SALES_ORDER_ITEM: fieldBuilder.buildEdmTypeField('SalesOrderItem', 'Edm.String', false),
            /**
             * Static representation of the [[docRelationshipUuid]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DOC_RELATIONSHIP_UUID: fieldBuilder.buildEdmTypeField('DocRelationshipUUID', 'Edm.Guid', false),
            /**
             * Static representation of the [[precedingDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PRECEDING_DOCUMENT: fieldBuilder.buildEdmTypeField('PrecedingDocument', 'Edm.String', true),
            /**
             * Static representation of the [[precedingDocumentItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PRECEDING_DOCUMENT_ITEM: fieldBuilder.buildEdmTypeField('PrecedingDocumentItem', 'Edm.String', true),
            /**
             * Static representation of the [[precedingDocumentCategory]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PRECEDING_DOCUMENT_CATEGORY: fieldBuilder.buildEdmTypeField('PrecedingDocumentCategory', 'Edm.String', true),
            /**
             * Static representation of the [[processFlowLevel]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROCESS_FLOW_LEVEL: fieldBuilder.buildEdmTypeField('ProcessFlowLevel', 'Edm.String', true),
            /**
             * Static representation of the [[creationDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CREATION_DATE: fieldBuilder.buildEdmTypeField('CreationDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[creationTime]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CREATION_TIME: fieldBuilder.buildEdmTypeField('CreationTime', 'Edm.Time', true),
            /**
             * Static representation of the [[lastChangeDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LAST_CHANGE_DATE: fieldBuilder.buildEdmTypeField('LastChangeDate', 'Edm.DateTime', true),
            ...this.navigationPropertyFields,
            /**
             *
             * All fields selector.
             */
            ALL_FIELDS: new odata_v2_1.AllFields('*', SalesOrderItmPrecdgProcFlow_1.SalesOrderItmPrecdgProcFlow)
        };
    }
}
exports.SalesOrderItmPrecdgProcFlowApi = SalesOrderItmPrecdgProcFlowApi;
