// @sap/cloud-sdk-vdm-sales-order-service/SalesOrderPartnerAddressApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesOrderPartnerAddressApi = void 0;
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This is a generated file powered by the SAP Cloud SDK for JavaScript.
 */
const SalesOrderPartnerAddress_1 = require("./SalesOrderPartnerAddress");
const SalesOrderPartnerAddressRequestBuilder_1 = require("./SalesOrderPartnerAddressRequestBuilder");
const odata_v2_1 = require("@sap-cloud-sdk/odata-v2");
class SalesOrderPartnerAddressApi {
    constructor(deSerializers = odata_v2_1.defaultDeSerializers) {
        this.entityConstructor = SalesOrderPartnerAddress_1.SalesOrderPartnerAddress;
        this.deSerializers = deSerializers;
    }
    _addNavigationProperties(linkedApis) {
        this.navigationPropertyFields = {
            TO_PARTNER: new odata_v2_1.OneToOneLink('to_Partner', this, linkedApis[0]),
            TO_SALES_ORDER: new odata_v2_1.OneToOneLink('to_SalesOrder', this, linkedApis[1])
        };
        return this;
    }
    requestBuilder() {
        return new SalesOrderPartnerAddressRequestBuilder_1.SalesOrderPartnerAddressRequestBuilder(this);
    }
    entityBuilder() {
        return (0, odata_v2_1.entityBuilder)(this);
    }
    customField(fieldName, isNullable = false) {
        return new odata_v2_1.CustomField(fieldName, this.entityConstructor, this.deSerializers, isNullable);
    }
    get schema() {
        const fieldBuilder = new odata_v2_1.FieldBuilder(SalesOrderPartnerAddress_1.SalesOrderPartnerAddress, this.deSerializers);
        return {
            /**
         * Static representation of the [[salesOrder]] property for query construction.
         * Use to reference this property in query operations such as 'select' in the fluent request API.
         */
            SALES_ORDER: fieldBuilder.buildEdmTypeField('SalesOrder', 'Edm.String', false),
            /**
             * Static representation of the [[partnerFunction]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_FUNCTION: fieldBuilder.buildEdmTypeField('PartnerFunction', 'Edm.String', false),
            /**
             * Static representation of the [[addressRepresentationCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ADDRESS_REPRESENTATION_CODE: fieldBuilder.buildEdmTypeField('AddressRepresentationCode', 'Edm.String', false),
            /**
             * Static representation of the [[correspondenceLanguage]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CORRESPONDENCE_LANGUAGE: fieldBuilder.buildEdmTypeField('CorrespondenceLanguage', 'Edm.String', true),
            /**
             * Static representation of the [[addresseeFullName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ADDRESSEE_FULL_NAME: fieldBuilder.buildEdmTypeField('AddresseeFullName', 'Edm.String', true),
            /**
             * Static representation of the [[organizationName1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORGANIZATION_NAME_1: fieldBuilder.buildEdmTypeField('OrganizationName1', 'Edm.String', true),
            /**
             * Static representation of the [[organizationName2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORGANIZATION_NAME_2: fieldBuilder.buildEdmTypeField('OrganizationName2', 'Edm.String', true),
            /**
             * Static representation of the [[organizationName3]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORGANIZATION_NAME_3: fieldBuilder.buildEdmTypeField('OrganizationName3', 'Edm.String', true),
            /**
             * Static representation of the [[organizationName4]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORGANIZATION_NAME_4: fieldBuilder.buildEdmTypeField('OrganizationName4', 'Edm.String', true),
            /**
             * Static representation of the [[cityName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CITY_NAME: fieldBuilder.buildEdmTypeField('CityName', 'Edm.String', true),
            /**
             * Static representation of the [[districtName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DISTRICT_NAME: fieldBuilder.buildEdmTypeField('DistrictName', 'Edm.String', true),
            /**
             * Static representation of the [[postalCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            POSTAL_CODE: fieldBuilder.buildEdmTypeField('PostalCode', 'Edm.String', true),
            /**
             * Static representation of the [[streetPrefixName1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            STREET_PREFIX_NAME_1: fieldBuilder.buildEdmTypeField('StreetPrefixName1', 'Edm.String', true),
            /**
             * Static representation of the [[streetPrefixName2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            STREET_PREFIX_NAME_2: fieldBuilder.buildEdmTypeField('StreetPrefixName2', 'Edm.String', true),
            /**
             * Static representation of the [[streetName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            STREET_NAME: fieldBuilder.buildEdmTypeField('StreetName', 'Edm.String', true),
            /**
             * Static representation of the [[streetSuffixName1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            STREET_SUFFIX_NAME_1: fieldBuilder.buildEdmTypeField('StreetSuffixName1', 'Edm.String', true),
            /**
             * Static representation of the [[streetSuffixName2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            STREET_SUFFIX_NAME_2: fieldBuilder.buildEdmTypeField('StreetSuffixName2', 'Edm.String', true),
            /**
             * Static representation of the [[houseNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            HOUSE_NUMBER: fieldBuilder.buildEdmTypeField('HouseNumber', 'Edm.String', true),
            /**
             * Static representation of the [[country]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COUNTRY: fieldBuilder.buildEdmTypeField('Country', 'Edm.String', true),
            /**
             * Static representation of the [[region]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REGION: fieldBuilder.buildEdmTypeField('Region', 'Edm.String', true),
            /**
             * Static representation of the [[formOfAddress]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FORM_OF_ADDRESS: fieldBuilder.buildEdmTypeField('FormOfAddress', 'Edm.String', true),
            /**
             * Static representation of the [[taxJurisdiction]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_JURISDICTION: fieldBuilder.buildEdmTypeField('TaxJurisdiction', 'Edm.String', true),
            /**
             * Static representation of the [[transportZone]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TRANSPORT_ZONE: fieldBuilder.buildEdmTypeField('TransportZone', 'Edm.String', true),
            /**
             * Static representation of the [[poBox]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PO_BOX: fieldBuilder.buildEdmTypeField('POBox', 'Edm.String', true),
            /**
             * Static representation of the [[poBoxPostalCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PO_BOX_POSTAL_CODE: fieldBuilder.buildEdmTypeField('POBoxPostalCode', 'Edm.String', true),
            /**
             * Static representation of the [[emailAddress]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            EMAIL_ADDRESS: fieldBuilder.buildEdmTypeField('EmailAddress', 'Edm.String', true),
            /**
             * Static representation of the [[mobilePhoneCountry]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            MOBILE_PHONE_COUNTRY: fieldBuilder.buildEdmTypeField('MobilePhoneCountry', 'Edm.String', true),
            /**
             * Static representation of the [[mobileNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            MOBILE_NUMBER: fieldBuilder.buildEdmTypeField('MobileNumber', 'Edm.String', true),
            /**
             * Static representation of the [[phoneNumberCountry]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PHONE_NUMBER_COUNTRY: fieldBuilder.buildEdmTypeField('PhoneNumberCountry', 'Edm.String', true),
            /**
             * Static representation of the [[phoneNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PHONE_NUMBER: fieldBuilder.buildEdmTypeField('PhoneNumber', 'Edm.String', true),
            /**
             * Static representation of the [[phoneExtensionNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PHONE_EXTENSION_NUMBER: fieldBuilder.buildEdmTypeField('PhoneExtensionNumber', 'Edm.String', true),
            /**
             * Static representation of the [[faxNumberCountry]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FAX_NUMBER_COUNTRY: fieldBuilder.buildEdmTypeField('FaxNumberCountry', 'Edm.String', true),
            /**
             * Static representation of the [[faxAreaCodeSubscriberNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FAX_AREA_CODE_SUBSCRIBER_NUMBER: fieldBuilder.buildEdmTypeField('FaxAreaCodeSubscriberNumber', 'Edm.String', true),
            /**
             * Static representation of the [[faxExtensionNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FAX_EXTENSION_NUMBER: fieldBuilder.buildEdmTypeField('FaxExtensionNumber', 'Edm.String', true),
            ...this.navigationPropertyFields,
            /**
             *
             * All fields selector.
             */
            ALL_FIELDS: new odata_v2_1.AllFields('*', SalesOrderPartnerAddress_1.SalesOrderPartnerAddress)
        };
    }
}
exports.SalesOrderPartnerAddressApi = SalesOrderPartnerAddressApi;
