// @sap/cloud-sdk-vdm-sales-order-service/SalesOrderBillingPlanItemApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesOrderBillingPlanItemApi = void 0;
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This is a generated file powered by the SAP Cloud SDK for JavaScript.
 */
const SalesOrderBillingPlanItem_1 = require("./SalesOrderBillingPlanItem");
const SalesOrderBillingPlanItemRequestBuilder_1 = require("./SalesOrderBillingPlanItemRequestBuilder");
const odata_v2_1 = require("@sap-cloud-sdk/odata-v2");
class SalesOrderBillingPlanItemApi {
    constructor(deSerializers = odata_v2_1.defaultDeSerializers) {
        this.entityConstructor = SalesOrderBillingPlanItem_1.SalesOrderBillingPlanItem;
        this.deSerializers = deSerializers;
    }
    _addNavigationProperties(linkedApis) {
        this.navigationPropertyFields = {
            TO_BILLING_PLAN: new odata_v2_1.OneToOneLink('to_BillingPlan', this, linkedApis[0]),
            TO_SALES_ORDER: new odata_v2_1.OneToOneLink('to_SalesOrder', this, linkedApis[1])
        };
        return this;
    }
    requestBuilder() {
        return new SalesOrderBillingPlanItemRequestBuilder_1.SalesOrderBillingPlanItemRequestBuilder(this);
    }
    entityBuilder() {
        return (0, odata_v2_1.entityBuilder)(this);
    }
    customField(fieldName, isNullable = false) {
        return new odata_v2_1.CustomField(fieldName, this.entityConstructor, this.deSerializers, isNullable);
    }
    get schema() {
        const fieldBuilder = new odata_v2_1.FieldBuilder(SalesOrderBillingPlanItem_1.SalesOrderBillingPlanItem, this.deSerializers);
        return {
            /**
         * Static representation of the [[salesOrder]] property for query construction.
         * Use to reference this property in query operations such as 'select' in the fluent request API.
         */
            SALES_ORDER: fieldBuilder.buildEdmTypeField('SalesOrder', 'Edm.String', false),
            /**
             * Static representation of the [[billingPlan]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN: fieldBuilder.buildEdmTypeField('BillingPlan', 'Edm.String', false),
            /**
             * Static representation of the [[billingPlanItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_ITEM: fieldBuilder.buildEdmTypeField('BillingPlanItem', 'Edm.String', false),
            /**
             * Static representation of the [[billingPlanDateCategory]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_DATE_CATEGORY: fieldBuilder.buildEdmTypeField('BillingPlanDateCategory', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanBillingDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_BILLING_DATE: fieldBuilder.buildEdmTypeField('BillingPlanBillingDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[billingPlanAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_AMOUNT: fieldBuilder.buildEdmTypeField('BillingPlanAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[transactionCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TRANSACTION_CURRENCY: fieldBuilder.buildEdmTypeField('TransactionCurrency', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanAmountPercent]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_AMOUNT_PERCENT: fieldBuilder.buildEdmTypeField('BillingPlanAmountPercent', 'Edm.Decimal', true),
            /**
             * Static representation of the [[customerPaymentTerms]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CUSTOMER_PAYMENT_TERMS: fieldBuilder.buildEdmTypeField('CustomerPaymentTerms', 'Edm.String', true),
            /**
             * Static representation of the [[proposedBillingDocumentType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROPOSED_BILLING_DOCUMENT_TYPE: fieldBuilder.buildEdmTypeField('ProposedBillingDocumentType', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanDateDescriptionCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_DATE_DESCRIPTION_CODE: fieldBuilder.buildEdmTypeField('BillingPlanDateDescriptionCode', 'Edm.String', true),
            /**
             * Static representation of the [[billingBlockReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_BLOCK_REASON: fieldBuilder.buildEdmTypeField('BillingBlockReason', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanServiceStartDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_SERVICE_START_DATE: fieldBuilder.buildEdmTypeField('BillingPlanServiceStartDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[billingPlanServiceEndDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_SERVICE_END_DATE: fieldBuilder.buildEdmTypeField('BillingPlanServiceEndDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[billingPlanRelatedBillgStatus]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_RELATED_BILLG_STATUS: fieldBuilder.buildEdmTypeField('BillingPlanRelatedBillgStatus', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_TYPE: fieldBuilder.buildEdmTypeField('BillingPlanType', 'Edm.String', true),
            /**
             * Static representation of the [[adoptingBillingDateId]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ADOPTING_BILLING_DATE_ID: fieldBuilder.buildEdmTypeField('AdoptingBillingDateID', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanBillingRule]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_BILLING_RULE: fieldBuilder.buildEdmTypeField('BillingPlanBillingRule', 'Edm.String', true),
            /**
             * Static representation of the [[billingPlanMilestoneUsage]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_PLAN_MILESTONE_USAGE: fieldBuilder.buildEdmTypeField('BillingPlanMilestoneUsage', 'Edm.String', true),
            /**
             * Static representation of the [[billgPlnDteCorrectionRfndType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLG_PLN_DTE_CORRECTION_RFND_TYPE: fieldBuilder.buildEdmTypeField('BillgPlnDteCorrectionRfndType', 'Edm.String', true),
            /**
             * Static representation of the [[accountingExchangeRate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_EXCHANGE_RATE: fieldBuilder.buildEdmTypeField('AccountingExchangeRate', 'Edm.Decimal', true),
            /**
             * Static representation of the [[postponementReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            POSTPONEMENT_REASON: fieldBuilder.buildEdmTypeField('PostponementReason', 'Edm.String', true),
            ...this.navigationPropertyFields,
            /**
             *
             * All fields selector.
             */
            ALL_FIELDS: new odata_v2_1.AllFields('*', SalesOrderBillingPlanItem_1.SalesOrderBillingPlanItem)
        };
    }
}
exports.SalesOrderBillingPlanItemApi = SalesOrderBillingPlanItemApi;
