// @sap/cloud-sdk-vdm-accounting-document-service/OperationalAcctgDocItemCubeRequestBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationalAcctgDocItemCubeRequestBuilder = void 0;
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This is a generated file powered by the SAP Cloud SDK for JavaScript.
 */
const odata_v2_1 = require("@sap-cloud-sdk/odata-v2");
/**
 * Request builder class for operations supported on the [[OperationalAcctgDocItemCube]] entity.
 */
class OperationalAcctgDocItemCubeRequestBuilder extends odata_v2_1.RequestBuilder {
    /**
     * Returns a request builder for retrieving one `OperationalAcctgDocItemCube` entity based on its keys.
     * @param companyCode Key property. See [[OperationalAcctgDocItemCube.companyCode]].
     * @param fiscalYear Key property. See [[OperationalAcctgDocItemCube.fiscalYear]].
     * @param accountingDocument Key property. See [[OperationalAcctgDocItemCube.accountingDocument]].
     * @param accountingDocumentItem Key property. See [[OperationalAcctgDocItemCube.accountingDocumentItem]].
     * @returns A request builder for creating requests to retrieve one `OperationalAcctgDocItemCube` entity based on its keys.
     */
    getByKey(companyCode, fiscalYear, accountingDocument, accountingDocumentItem) {
        return new odata_v2_1.GetByKeyRequestBuilder(this.entityApi, {
            CompanyCode: companyCode,
            FiscalYear: fiscalYear,
            AccountingDocument: accountingDocument,
            AccountingDocumentItem: accountingDocumentItem
        });
    }
    /**
     * Returns a request builder for querying all `OperationalAcctgDocItemCube` entities.
     * @returns A request builder for creating requests to retrieve all `OperationalAcctgDocItemCube` entities.
     */
    getAll() {
        return new odata_v2_1.GetAllRequestBuilder(this.entityApi);
    }
}
exports.OperationalAcctgDocItemCubeRequestBuilder = OperationalAcctgDocItemCubeRequestBuilder;
