// @sap/cloud-sdk-vdm-accounting-document-service/OperationalAcctgDocItemCubeApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationalAcctgDocItemCubeApi = void 0;
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This is a generated file powered by the SAP Cloud SDK for JavaScript.
 */
const OperationalAcctgDocItemCube_1 = require("./OperationalAcctgDocItemCube");
const OperationalAcctgDocItemCubeRequestBuilder_1 = require("./OperationalAcctgDocItemCubeRequestBuilder");
const odata_v2_1 = require("@sap-cloud-sdk/odata-v2");
class OperationalAcctgDocItemCubeApi {
    constructor(deSerializers = odata_v2_1.defaultDeSerializers) {
        this.entityConstructor = OperationalAcctgDocItemCube_1.OperationalAcctgDocItemCube;
        this.deSerializers = deSerializers;
    }
    _addNavigationProperties(linkedApis) {
        this.navigationPropertyFields = {};
        return this;
    }
    requestBuilder() {
        return new OperationalAcctgDocItemCubeRequestBuilder_1.OperationalAcctgDocItemCubeRequestBuilder(this);
    }
    entityBuilder() {
        return (0, odata_v2_1.entityBuilder)(this);
    }
    customField(fieldName, isNullable = false) {
        return new odata_v2_1.CustomField(fieldName, this.entityConstructor, this.deSerializers, isNullable);
    }
    get schema() {
        const fieldBuilder = new odata_v2_1.FieldBuilder(OperationalAcctgDocItemCube_1.OperationalAcctgDocItemCube, this.deSerializers);
        return {
            /**
         * Static representation of the [[companyCode]] property for query construction.
         * Use to reference this property in query operations such as 'select' in the fluent request API.
         */
            COMPANY_CODE: fieldBuilder.buildEdmTypeField('CompanyCode', 'Edm.String', false),
            /**
             * Static representation of the [[fiscalYear]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FISCAL_YEAR: fieldBuilder.buildEdmTypeField('FiscalYear', 'Edm.String', false),
            /**
             * Static representation of the [[accountingDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT: fieldBuilder.buildEdmTypeField('AccountingDocument', 'Edm.String', false),
            /**
             * Static representation of the [[accountingDocumentItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_ITEM: fieldBuilder.buildEdmTypeField('AccountingDocumentItem', 'Edm.String', false),
            /**
             * Static representation of the [[companyCodeName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COMPANY_CODE_NAME: fieldBuilder.buildEdmTypeField('CompanyCodeName', 'Edm.String', true),
            /**
             * Static representation of the [[chartOfAccounts]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CHART_OF_ACCOUNTS: fieldBuilder.buildEdmTypeField('ChartOfAccounts', 'Edm.String', true),
            /**
             * Static representation of the [[accountingDocumentItemType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_ITEM_TYPE: fieldBuilder.buildEdmTypeField('AccountingDocumentItemType', 'Edm.String', true),
            /**
             * Static representation of the [[clearingDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CLEARING_DATE: fieldBuilder.buildEdmTypeField('ClearingDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[clearingCreationDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CLEARING_CREATION_DATE: fieldBuilder.buildEdmTypeField('ClearingCreationDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[clearingAccountingDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CLEARING_ACCOUNTING_DOCUMENT: fieldBuilder.buildEdmTypeField('ClearingAccountingDocument', 'Edm.String', true),
            /**
             * Static representation of the [[isCleared]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_CLEARED: fieldBuilder.buildEdmTypeField('IsCleared', 'Edm.Boolean', true),
            /**
             * Static representation of the [[postingKey]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            POSTING_KEY: fieldBuilder.buildEdmTypeField('PostingKey', 'Edm.String', true),
            /**
             * Static representation of the [[financialAccountType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FINANCIAL_ACCOUNT_TYPE: fieldBuilder.buildEdmTypeField('FinancialAccountType', 'Edm.String', true),
            /**
             * Static representation of the [[specialGlCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SPECIAL_GL_CODE: fieldBuilder.buildEdmTypeField('SpecialGLCode', 'Edm.String', true),
            /**
             * Static representation of the [[specialGlTransactionType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SPECIAL_GL_TRANSACTION_TYPE: fieldBuilder.buildEdmTypeField('SpecialGLTransactionType', 'Edm.String', true),
            /**
             * Static representation of the [[debitCreditCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DEBIT_CREDIT_CODE: fieldBuilder.buildEdmTypeField('DebitCreditCode', 'Edm.String', true),
            /**
             * Static representation of the [[businessArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BUSINESS_AREA: fieldBuilder.buildEdmTypeField('BusinessArea', 'Edm.String', true),
            /**
             * Static representation of the [[businessAreaName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BUSINESS_AREA_NAME: fieldBuilder.buildEdmTypeField('BusinessAreaName', 'Edm.String', true),
            /**
             * Static representation of the [[partnerBusinessArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_BUSINESS_AREA: fieldBuilder.buildEdmTypeField('PartnerBusinessArea', 'Edm.String', true),
            /**
             * Static representation of the [[taxCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_CODE: fieldBuilder.buildEdmTypeField('TaxCode', 'Edm.String', true),
            /**
             * Static representation of the [[withholdingTaxCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            WITHHOLDING_TAX_CODE: fieldBuilder.buildEdmTypeField('WithholdingTaxCode', 'Edm.String', true),
            /**
             * Static representation of the [[taxType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_TYPE: fieldBuilder.buildEdmTypeField('TaxType', 'Edm.String', true),
            /**
             * Static representation of the [[transactionTypeDetermination]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TRANSACTION_TYPE_DETERMINATION: fieldBuilder.buildEdmTypeField('TransactionTypeDetermination', 'Edm.String', true),
            /**
             * Static representation of the [[valueDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            VALUE_DATE: fieldBuilder.buildEdmTypeField('ValueDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[assignmentReference]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ASSIGNMENT_REFERENCE: fieldBuilder.buildEdmTypeField('AssignmentReference', 'Edm.String', true),
            /**
             * Static representation of the [[documentItemText]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DOCUMENT_ITEM_TEXT: fieldBuilder.buildEdmTypeField('DocumentItemText', 'Edm.String', true),
            /**
             * Static representation of the [[partnerCompany]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_COMPANY: fieldBuilder.buildEdmTypeField('PartnerCompany', 'Edm.String', true),
            /**
             * Static representation of the [[financialTransactionType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FINANCIAL_TRANSACTION_TYPE: fieldBuilder.buildEdmTypeField('FinancialTransactionType', 'Edm.String', true),
            /**
             * Static representation of the [[corporateGroupAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CORPORATE_GROUP_ACCOUNT: fieldBuilder.buildEdmTypeField('CorporateGroupAccount', 'Edm.String', true),
            /**
             * Static representation of the [[planningLevel]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PLANNING_LEVEL: fieldBuilder.buildEdmTypeField('PlanningLevel', 'Edm.String', true),
            /**
             * Static representation of the [[controllingArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CONTROLLING_AREA: fieldBuilder.buildEdmTypeField('ControllingArea', 'Edm.String', true),
            /**
             * Static representation of the [[controllingAreaName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CONTROLLING_AREA_NAME: fieldBuilder.buildEdmTypeField('ControllingAreaName', 'Edm.String', true),
            /**
             * Static representation of the [[costCenter]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COST_CENTER: fieldBuilder.buildEdmTypeField('CostCenter', 'Edm.String', true),
            /**
             * Static representation of the [[costCenterName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COST_CENTER_NAME: fieldBuilder.buildEdmTypeField('CostCenterName', 'Edm.String', true),
            /**
             * Static representation of the [[project]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROJECT: fieldBuilder.buildEdmTypeField('Project', 'Edm.String', true),
            /**
             * Static representation of the [[orderId]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORDER_ID: fieldBuilder.buildEdmTypeField('OrderID', 'Edm.String', true),
            /**
             * Static representation of the [[billingDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILLING_DOCUMENT: fieldBuilder.buildEdmTypeField('BillingDocument', 'Edm.String', true),
            /**
             * Static representation of the [[salesDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SALES_DOCUMENT: fieldBuilder.buildEdmTypeField('SalesDocument', 'Edm.String', true),
            /**
             * Static representation of the [[salesDocumentItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SALES_DOCUMENT_ITEM: fieldBuilder.buildEdmTypeField('SalesDocumentItem', 'Edm.String', true),
            /**
             * Static representation of the [[scheduleLine]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SCHEDULE_LINE: fieldBuilder.buildEdmTypeField('ScheduleLine', 'Edm.String', true),
            /**
             * Static representation of the [[masterFixedAsset]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            MASTER_FIXED_ASSET: fieldBuilder.buildEdmTypeField('MasterFixedAsset', 'Edm.String', true),
            /**
             * Static representation of the [[fixedAsset]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FIXED_ASSET: fieldBuilder.buildEdmTypeField('FixedAsset', 'Edm.String', true),
            /**
             * Static representation of the [[assetTransactionType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ASSET_TRANSACTION_TYPE: fieldBuilder.buildEdmTypeField('AssetTransactionType', 'Edm.String', true),
            /**
             * Static representation of the [[assetValueDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ASSET_VALUE_DATE: fieldBuilder.buildEdmTypeField('AssetValueDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[personnelNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PERSONNEL_NUMBER: fieldBuilder.buildEdmTypeField('PersonnelNumber', 'Edm.String', true),
            /**
             * Static representation of the [[isSalesRelated]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_SALES_RELATED: fieldBuilder.buildEdmTypeField('IsSalesRelated', 'Edm.Boolean', true),
            /**
             * Static representation of the [[lineItemDisplayIsEnabled]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LINE_ITEM_DISPLAY_IS_ENABLED: fieldBuilder.buildEdmTypeField('LineItemDisplayIsEnabled', 'Edm.Boolean', true),
            /**
             * Static representation of the [[isOpenItemManaged]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_OPEN_ITEM_MANAGED: fieldBuilder.buildEdmTypeField('IsOpenItemManaged', 'Edm.String', true),
            /**
             * Static representation of the [[isNotCashDiscountLiable]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_NOT_CASH_DISCOUNT_LIABLE: fieldBuilder.buildEdmTypeField('IsNotCashDiscountLiable', 'Edm.Boolean', true),
            /**
             * Static representation of the [[isAutomaticallyCreated]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_AUTOMATICALLY_CREATED: fieldBuilder.buildEdmTypeField('IsAutomaticallyCreated', 'Edm.Boolean', true),
            /**
             * Static representation of the [[isUsedInPaymentTransaction]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_USED_IN_PAYMENT_TRANSACTION: fieldBuilder.buildEdmTypeField('IsUsedInPaymentTransaction', 'Edm.Boolean', true),
            /**
             * Static representation of the [[operationalGlAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            OPERATIONAL_GL_ACCOUNT: fieldBuilder.buildEdmTypeField('OperationalGLAccount', 'Edm.String', true),
            /**
             * Static representation of the [[glAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            GL_ACCOUNT: fieldBuilder.buildEdmTypeField('GLAccount', 'Edm.String', true),
            /**
             * Static representation of the [[glAccountName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            GL_ACCOUNT_NAME: fieldBuilder.buildEdmTypeField('GLAccountName', 'Edm.String', true),
            /**
             * Static representation of the [[glAccountLongName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            GL_ACCOUNT_LONG_NAME: fieldBuilder.buildEdmTypeField('GLAccountLongName', 'Edm.String', true),
            /**
             * Static representation of the [[customer]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CUSTOMER: fieldBuilder.buildEdmTypeField('Customer', 'Edm.String', true),
            /**
             * Static representation of the [[customerName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CUSTOMER_NAME: fieldBuilder.buildEdmTypeField('CustomerName', 'Edm.String', true),
            /**
             * Static representation of the [[supplier]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SUPPLIER: fieldBuilder.buildEdmTypeField('Supplier', 'Edm.String', true),
            /**
             * Static representation of the [[supplierName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SUPPLIER_NAME: fieldBuilder.buildEdmTypeField('SupplierName', 'Edm.String', true),
            /**
             * Static representation of the [[branchAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BRANCH_ACCOUNT: fieldBuilder.buildEdmTypeField('BranchAccount', 'Edm.String', true),
            /**
             * Static representation of the [[isBalanceSheetAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_BALANCE_SHEET_ACCOUNT: fieldBuilder.buildEdmTypeField('IsBalanceSheetAccount', 'Edm.Boolean', true),
            /**
             * Static representation of the [[profitLossAccountType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROFIT_LOSS_ACCOUNT_TYPE: fieldBuilder.buildEdmTypeField('ProfitLossAccountType', 'Edm.String', true),
            /**
             * Static representation of the [[specialGlAccountAssignment]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SPECIAL_GL_ACCOUNT_ASSIGNMENT: fieldBuilder.buildEdmTypeField('SpecialGLAccountAssignment', 'Edm.String', true),
            /**
             * Static representation of the [[dueCalculationBaseDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DUE_CALCULATION_BASE_DATE: fieldBuilder.buildEdmTypeField('DueCalculationBaseDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[paymentTerms]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_TERMS: fieldBuilder.buildEdmTypeField('PaymentTerms', 'Edm.String', true),
            /**
             * Static representation of the [[cashDiscount1Days]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_1_DAYS: fieldBuilder.buildEdmTypeField('CashDiscount1Days', 'Edm.Decimal', true),
            /**
             * Static representation of the [[cashDiscount2Days]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_2_DAYS: fieldBuilder.buildEdmTypeField('CashDiscount2Days', 'Edm.Decimal', true),
            /**
             * Static representation of the [[netPaymentDays]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            NET_PAYMENT_DAYS: fieldBuilder.buildEdmTypeField('NetPaymentDays', 'Edm.Decimal', true),
            /**
             * Static representation of the [[cashDiscount1Percent]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_1_PERCENT: fieldBuilder.buildEdmTypeField('CashDiscount1Percent', 'Edm.Decimal', true),
            /**
             * Static representation of the [[cashDiscount2Percent]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_2_PERCENT: fieldBuilder.buildEdmTypeField('CashDiscount2Percent', 'Edm.Decimal', true),
            /**
             * Static representation of the [[paymentMethod]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_METHOD: fieldBuilder.buildEdmTypeField('PaymentMethod', 'Edm.String', true),
            /**
             * Static representation of the [[paymentBlockingReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_BLOCKING_REASON: fieldBuilder.buildEdmTypeField('PaymentBlockingReason', 'Edm.String', true),
            /**
             * Static representation of the [[fixedCashDiscount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FIXED_CASH_DISCOUNT: fieldBuilder.buildEdmTypeField('FixedCashDiscount', 'Edm.String', true),
            /**
             * Static representation of the [[houseBank]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            HOUSE_BANK: fieldBuilder.buildEdmTypeField('HouseBank', 'Edm.String', true),
            /**
             * Static representation of the [[bpBankAccountInternalId]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BP_BANK_ACCOUNT_INTERNAL_ID: fieldBuilder.buildEdmTypeField('BPBankAccountInternalID', 'Edm.String', true),
            /**
             * Static representation of the [[taxDistributionCode1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_DISTRIBUTION_CODE_1: fieldBuilder.buildEdmTypeField('TaxDistributionCode1', 'Edm.String', true),
            /**
             * Static representation of the [[taxDistributionCode2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_DISTRIBUTION_CODE_2: fieldBuilder.buildEdmTypeField('TaxDistributionCode2', 'Edm.String', true),
            /**
             * Static representation of the [[taxDistributionCode3]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_DISTRIBUTION_CODE_3: fieldBuilder.buildEdmTypeField('TaxDistributionCode3', 'Edm.String', true),
            /**
             * Static representation of the [[invoiceReference]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_REFERENCE: fieldBuilder.buildEdmTypeField('InvoiceReference', 'Edm.String', true),
            /**
             * Static representation of the [[invoiceReferenceFiscalYear]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_REFERENCE_FISCAL_YEAR: fieldBuilder.buildEdmTypeField('InvoiceReferenceFiscalYear', 'Edm.String', true),
            /**
             * Static representation of the [[invoiceItemReference]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_ITEM_REFERENCE: fieldBuilder.buildEdmTypeField('InvoiceItemReference', 'Edm.String', true),
            /**
             * Static representation of the [[followOnDocumentType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FOLLOW_ON_DOCUMENT_TYPE: fieldBuilder.buildEdmTypeField('FollowOnDocumentType', 'Edm.String', true),
            /**
             * Static representation of the [[stateCentralBankPaymentReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            STATE_CENTRAL_BANK_PAYMENT_REASON: fieldBuilder.buildEdmTypeField('StateCentralBankPaymentReason', 'Edm.String', true),
            /**
             * Static representation of the [[supplyingCountry]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SUPPLYING_COUNTRY: fieldBuilder.buildEdmTypeField('SupplyingCountry', 'Edm.String', true),
            /**
             * Static representation of the [[invoiceList]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_LIST: fieldBuilder.buildEdmTypeField('InvoiceList', 'Edm.String', true),
            /**
             * Static representation of the [[billOfExchangeUsage]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BILL_OF_EXCHANGE_USAGE: fieldBuilder.buildEdmTypeField('BillOfExchangeUsage', 'Edm.String', true),
            /**
             * Static representation of the [[dunningKey]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DUNNING_KEY: fieldBuilder.buildEdmTypeField('DunningKey', 'Edm.String', true),
            /**
             * Static representation of the [[dunningBlockingReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DUNNING_BLOCKING_REASON: fieldBuilder.buildEdmTypeField('DunningBlockingReason', 'Edm.String', true),
            /**
             * Static representation of the [[lastDunningDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LAST_DUNNING_DATE: fieldBuilder.buildEdmTypeField('LastDunningDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[dunningLevel]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DUNNING_LEVEL: fieldBuilder.buildEdmTypeField('DunningLevel', 'Edm.String', true),
            /**
             * Static representation of the [[dunningArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DUNNING_AREA: fieldBuilder.buildEdmTypeField('DunningArea', 'Edm.String', true),
            /**
             * Static representation of the [[withholdingTaxCertificate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            WITHHOLDING_TAX_CERTIFICATE: fieldBuilder.buildEdmTypeField('WithholdingTaxCertificate', 'Edm.String', true),
            /**
             * Static representation of the [[material]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            MATERIAL: fieldBuilder.buildEdmTypeField('Material', 'Edm.String', true),
            /**
             * Static representation of the [[product]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PRODUCT: fieldBuilder.buildEdmTypeField('Product', 'Edm.String', true),
            /**
             * Static representation of the [[plant]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PLANT: fieldBuilder.buildEdmTypeField('Plant', 'Edm.String', true),
            /**
             * Static representation of the [[purchasingDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PURCHASING_DOCUMENT: fieldBuilder.buildEdmTypeField('PurchasingDocument', 'Edm.String', true),
            /**
             * Static representation of the [[purchasingDocumentItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PURCHASING_DOCUMENT_ITEM: fieldBuilder.buildEdmTypeField('PurchasingDocumentItem', 'Edm.String', true),
            /**
             * Static representation of the [[accountAssignmentNumber]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNT_ASSIGNMENT_NUMBER: fieldBuilder.buildEdmTypeField('AccountAssignmentNumber', 'Edm.String', true),
            /**
             * Static representation of the [[isCompletelyDelivered]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_COMPLETELY_DELIVERED: fieldBuilder.buildEdmTypeField('IsCompletelyDelivered', 'Edm.Boolean', true),
            /**
             * Static representation of the [[materialPriceControl]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            MATERIAL_PRICE_CONTROL: fieldBuilder.buildEdmTypeField('MaterialPriceControl', 'Edm.String', true),
            /**
             * Static representation of the [[valuationArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            VALUATION_AREA: fieldBuilder.buildEdmTypeField('ValuationArea', 'Edm.String', true),
            /**
             * Static representation of the [[inventoryValuationType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVENTORY_VALUATION_TYPE: fieldBuilder.buildEdmTypeField('InventoryValuationType', 'Edm.String', true),
            /**
             * Static representation of the [[vatRegistration]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            VAT_REGISTRATION: fieldBuilder.buildEdmTypeField('VATRegistration', 'Edm.String', true),
            /**
             * Static representation of the [[delivOfGoodsDestCountry]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DELIV_OF_GOODS_DEST_COUNTRY: fieldBuilder.buildEdmTypeField('DelivOfGoodsDestCountry', 'Edm.String', true),
            /**
             * Static representation of the [[paymentDifferenceReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_DIFFERENCE_REASON: fieldBuilder.buildEdmTypeField('PaymentDifferenceReason', 'Edm.String', true),
            /**
             * Static representation of the [[profitCenter]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROFIT_CENTER: fieldBuilder.buildEdmTypeField('ProfitCenter', 'Edm.String', true),
            /**
             * Static representation of the [[profitCenterName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROFIT_CENTER_NAME: fieldBuilder.buildEdmTypeField('ProfitCenterName', 'Edm.String', true),
            /**
             * Static representation of the [[jointVenture]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            JOINT_VENTURE: fieldBuilder.buildEdmTypeField('JointVenture', 'Edm.String', true),
            /**
             * Static representation of the [[jointVentureCostRecoveryCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            JOINT_VENTURE_COST_RECOVERY_CODE: fieldBuilder.buildEdmTypeField('JointVentureCostRecoveryCode', 'Edm.String', true),
            /**
             * Static representation of the [[jointVentureEquityGroup]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            JOINT_VENTURE_EQUITY_GROUP: fieldBuilder.buildEdmTypeField('JointVentureEquityGroup', 'Edm.String', true),
            /**
             * Static representation of the [[treasuryContractType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TREASURY_CONTRACT_TYPE: fieldBuilder.buildEdmTypeField('TreasuryContractType', 'Edm.String', true),
            /**
             * Static representation of the [[assetContract]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ASSET_CONTRACT: fieldBuilder.buildEdmTypeField('AssetContract', 'Edm.String', true),
            /**
             * Static representation of the [[cashFlowType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_FLOW_TYPE: fieldBuilder.buildEdmTypeField('CashFlowType', 'Edm.String', true),
            /**
             * Static representation of the [[taxJurisdiction]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_JURISDICTION: fieldBuilder.buildEdmTypeField('TaxJurisdiction', 'Edm.String', true),
            /**
             * Static representation of the [[realEstateObject]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REAL_ESTATE_OBJECT: fieldBuilder.buildEdmTypeField('RealEstateObject', 'Edm.String', true),
            /**
             * Static representation of the [[settlementReferenceDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SETTLEMENT_REFERENCE_DATE: fieldBuilder.buildEdmTypeField('SettlementReferenceDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[commitmentItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COMMITMENT_ITEM: fieldBuilder.buildEdmTypeField('CommitmentItem', 'Edm.String', true),
            /**
             * Static representation of the [[costObject]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COST_OBJECT: fieldBuilder.buildEdmTypeField('CostObject', 'Edm.String', true),
            /**
             * Static representation of the [[projectNetwork]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROJECT_NETWORK: fieldBuilder.buildEdmTypeField('ProjectNetwork', 'Edm.String', true),
            /**
             * Static representation of the [[orderInternalBillOfOperations]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORDER_INTERNAL_BILL_OF_OPERATIONS: fieldBuilder.buildEdmTypeField('OrderInternalBillOfOperations', 'Edm.String', true),
            /**
             * Static representation of the [[orderIntBillOfOperationsItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORDER_INT_BILL_OF_OPERATIONS_ITEM: fieldBuilder.buildEdmTypeField('OrderIntBillOfOperationsItem', 'Edm.String', true),
            /**
             * Static representation of the [[wbsElementInternalId]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            WBS_ELEMENT_INTERNAL_ID: fieldBuilder.buildEdmTypeField('WBSElementInternalID', 'Edm.String', true),
            /**
             * Static representation of the [[profitabilitySegment]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PROFITABILITY_SEGMENT: fieldBuilder.buildEdmTypeField('ProfitabilitySegment', 'Edm.String', true),
            /**
             * Static representation of the [[jointVentureEquityType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            JOINT_VENTURE_EQUITY_TYPE: fieldBuilder.buildEdmTypeField('JointVentureEquityType', 'Edm.String', true),
            /**
             * Static representation of the [[isEuTriangularDeal]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_EU_TRIANGULAR_DEAL: fieldBuilder.buildEdmTypeField('IsEUTriangularDeal', 'Edm.Boolean', true),
            /**
             * Static representation of the [[costOriginGroup]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COST_ORIGIN_GROUP: fieldBuilder.buildEdmTypeField('CostOriginGroup', 'Edm.String', true),
            /**
             * Static representation of the [[companyCodeCurrencyDetnMethod]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COMPANY_CODE_CURRENCY_DETN_METHOD: fieldBuilder.buildEdmTypeField('CompanyCodeCurrencyDetnMethod', 'Edm.String', true),
            /**
             * Static representation of the [[clearingIsReversed]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CLEARING_IS_REVERSED: fieldBuilder.buildEdmTypeField('ClearingIsReversed', 'Edm.Boolean', true),
            /**
             * Static representation of the [[paymentMethodSupplement]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_METHOD_SUPPLEMENT: fieldBuilder.buildEdmTypeField('PaymentMethodSupplement', 'Edm.String', true),
            /**
             * Static representation of the [[alternativeGlAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ALTERNATIVE_GL_ACCOUNT: fieldBuilder.buildEdmTypeField('AlternativeGLAccount', 'Edm.String', true),
            /**
             * Static representation of the [[fundsCenter]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FUNDS_CENTER: fieldBuilder.buildEdmTypeField('FundsCenter', 'Edm.String', true),
            /**
             * Static representation of the [[fund]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FUND: fieldBuilder.buildEdmTypeField('Fund', 'Edm.String', true),
            /**
             * Static representation of the [[partnerProfitCenter]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_PROFIT_CENTER: fieldBuilder.buildEdmTypeField('PartnerProfitCenter', 'Edm.String', true),
            /**
             * Static representation of the [[reference1IdByBusinessPartner]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_1_ID_BY_BUSINESS_PARTNER: fieldBuilder.buildEdmTypeField('Reference1IDByBusinessPartner', 'Edm.String', true),
            /**
             * Static representation of the [[reference2IdByBusinessPartner]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_2_ID_BY_BUSINESS_PARTNER: fieldBuilder.buildEdmTypeField('Reference2IDByBusinessPartner', 'Edm.String', true),
            /**
             * Static representation of the [[isNegativePosting]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_NEGATIVE_POSTING: fieldBuilder.buildEdmTypeField('IsNegativePosting', 'Edm.Boolean', true),
            /**
             * Static representation of the [[paymentCardItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_CARD_ITEM: fieldBuilder.buildEdmTypeField('PaymentCardItem', 'Edm.String', true),
            /**
             * Static representation of the [[paymentCardPaymentSettlement]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_CARD_PAYMENT_SETTLEMENT: fieldBuilder.buildEdmTypeField('PaymentCardPaymentSettlement', 'Edm.String', true),
            /**
             * Static representation of the [[creditControlArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CREDIT_CONTROL_AREA: fieldBuilder.buildEdmTypeField('CreditControlArea', 'Edm.String', true),
            /**
             * Static representation of the [[reference3IdByBusinessPartner]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_3_ID_BY_BUSINESS_PARTNER: fieldBuilder.buildEdmTypeField('Reference3IDByBusinessPartner', 'Edm.String', true),
            /**
             * Static representation of the [[dataExchangeInstruction1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DATA_EXCHANGE_INSTRUCTION_1: fieldBuilder.buildEdmTypeField('DataExchangeInstruction1', 'Edm.String', true),
            /**
             * Static representation of the [[dataExchangeInstruction2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DATA_EXCHANGE_INSTRUCTION_2: fieldBuilder.buildEdmTypeField('DataExchangeInstruction2', 'Edm.String', true),
            /**
             * Static representation of the [[dataExchangeInstruction3]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DATA_EXCHANGE_INSTRUCTION_3: fieldBuilder.buildEdmTypeField('DataExchangeInstruction3', 'Edm.String', true),
            /**
             * Static representation of the [[dataExchangeInstruction4]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DATA_EXCHANGE_INSTRUCTION_4: fieldBuilder.buildEdmTypeField('DataExchangeInstruction4', 'Edm.String', true),
            /**
             * Static representation of the [[region]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REGION: fieldBuilder.buildEdmTypeField('Region', 'Edm.String', true),
            /**
             * Static representation of the [[hasPaymentOrder]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            HAS_PAYMENT_ORDER: fieldBuilder.buildEdmTypeField('HasPaymentOrder', 'Edm.Boolean', true),
            /**
             * Static representation of the [[paymentReference]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_REFERENCE: fieldBuilder.buildEdmTypeField('PaymentReference', 'Edm.String', true),
            /**
             * Static representation of the [[taxDeterminationDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_DETERMINATION_DATE: fieldBuilder.buildEdmTypeField('TaxDeterminationDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[clearingItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CLEARING_ITEM: fieldBuilder.buildEdmTypeField('ClearingItem', 'Edm.Decimal', true),
            /**
             * Static representation of the [[businessPlace]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BUSINESS_PLACE: fieldBuilder.buildEdmTypeField('BusinessPlace', 'Edm.String', true),
            /**
             * Static representation of the [[taxSection]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_SECTION: fieldBuilder.buildEdmTypeField('TaxSection', 'Edm.String', true),
            /**
             * Static representation of the [[costCtrActivityType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COST_CTR_ACTIVITY_TYPE: fieldBuilder.buildEdmTypeField('CostCtrActivityType', 'Edm.String', true),
            /**
             * Static representation of the [[accountsReceivableIsPledged]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTS_RECEIVABLE_IS_PLEDGED: fieldBuilder.buildEdmTypeField('AccountsReceivableIsPledged', 'Edm.String', true),
            /**
             * Static representation of the [[businessProcess]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BUSINESS_PROCESS: fieldBuilder.buildEdmTypeField('BusinessProcess', 'Edm.String', true),
            /**
             * Static representation of the [[grantId]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            GRANT_ID: fieldBuilder.buildEdmTypeField('GrantID', 'Edm.String', true),
            /**
             * Static representation of the [[functionalArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FUNCTIONAL_AREA: fieldBuilder.buildEdmTypeField('FunctionalArea', 'Edm.String', true),
            /**
             * Static representation of the [[functionalAreaName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FUNCTIONAL_AREA_NAME: fieldBuilder.buildEdmTypeField('FunctionalAreaName', 'Edm.String', true),
            /**
             * Static representation of the [[customerIsInExecution]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CUSTOMER_IS_IN_EXECUTION: fieldBuilder.buildEdmTypeField('CustomerIsInExecution', 'Edm.Boolean', true),
            /**
             * Static representation of the [[fundedProgram]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FUNDED_PROGRAM: fieldBuilder.buildEdmTypeField('FundedProgram', 'Edm.String', true),
            /**
             * Static representation of the [[clearingDocFiscalYear]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CLEARING_DOC_FISCAL_YEAR: fieldBuilder.buildEdmTypeField('ClearingDocFiscalYear', 'Edm.String', true),
            /**
             * Static representation of the [[ledgerGlLineItem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LEDGER_GL_LINE_ITEM: fieldBuilder.buildEdmTypeField('LedgerGLLineItem', 'Edm.String', true),
            /**
             * Static representation of the [[segment]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SEGMENT: fieldBuilder.buildEdmTypeField('Segment', 'Edm.String', true),
            /**
             * Static representation of the [[segmentName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SEGMENT_NAME: fieldBuilder.buildEdmTypeField('SegmentName', 'Edm.String', true),
            /**
             * Static representation of the [[partnerSegment]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_SEGMENT: fieldBuilder.buildEdmTypeField('PartnerSegment', 'Edm.String', true),
            /**
             * Static representation of the [[partnerFunctionalArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_FUNCTIONAL_AREA: fieldBuilder.buildEdmTypeField('PartnerFunctionalArea', 'Edm.String', true),
            /**
             * Static representation of the [[houseBankAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            HOUSE_BANK_ACCOUNT: fieldBuilder.buildEdmTypeField('HouseBankAccount', 'Edm.String', true),
            /**
             * Static representation of the [[costElement]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COST_ELEMENT: fieldBuilder.buildEdmTypeField('CostElement', 'Edm.String', true),
            /**
             * Static representation of the [[sepaMandate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SEPA_MANDATE: fieldBuilder.buildEdmTypeField('SEPAMandate', 'Edm.String', true),
            /**
             * Static representation of the [[referenceDocumentType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_DOCUMENT_TYPE: fieldBuilder.buildEdmTypeField('ReferenceDocumentType', 'Edm.String', true),
            /**
             * Static representation of the [[originalReferenceDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORIGINAL_REFERENCE_DOCUMENT: fieldBuilder.buildEdmTypeField('OriginalReferenceDocument', 'Edm.String', true),
            /**
             * Static representation of the [[referenceDocumentLogicalSystem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_DOCUMENT_LOGICAL_SYSTEM: fieldBuilder.buildEdmTypeField('ReferenceDocumentLogicalSystem', 'Edm.String', true),
            /**
             * Static representation of the [[accountingDocumentItemRef]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_ITEM_REF: fieldBuilder.buildEdmTypeField('AccountingDocumentItemRef', 'Edm.String', true),
            /**
             * Static representation of the [[fiscalPeriod]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FISCAL_PERIOD: fieldBuilder.buildEdmTypeField('FiscalPeriod', 'Edm.String', true),
            /**
             * Static representation of the [[accountingDocumentCategory]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_CATEGORY: fieldBuilder.buildEdmTypeField('AccountingDocumentCategory', 'Edm.String', true),
            /**
             * Static representation of the [[accountingDocumentCategoryName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_CATEGORY_NAME: fieldBuilder.buildEdmTypeField('AccountingDocumentCategoryName', 'Edm.String', true),
            /**
             * Static representation of the [[postingDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            POSTING_DATE: fieldBuilder.buildEdmTypeField('PostingDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[documentDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DOCUMENT_DATE: fieldBuilder.buildEdmTypeField('DocumentDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[accountingDocumentType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_TYPE: fieldBuilder.buildEdmTypeField('AccountingDocumentType', 'Edm.String', true),
            /**
             * Static representation of the [[accountingDocumentTypeName]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_TYPE_NAME: fieldBuilder.buildEdmTypeField('AccountingDocumentTypeName', 'Edm.String', true),
            /**
             * Static representation of the [[netDueDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            NET_DUE_DATE: fieldBuilder.buildEdmTypeField('NetDueDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[cashDiscount1DueDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_1_DUE_DATE: fieldBuilder.buildEdmTypeField('CashDiscount1DueDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[cashDiscount2DueDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_2_DUE_DATE: fieldBuilder.buildEdmTypeField('CashDiscount2DueDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[offsettingAccount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            OFFSETTING_ACCOUNT: fieldBuilder.buildEdmTypeField('OffsettingAccount', 'Edm.String', true),
            /**
             * Static representation of the [[offsettingAccountType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            OFFSETTING_ACCOUNT_TYPE: fieldBuilder.buildEdmTypeField('OffsettingAccountType', 'Edm.String', true),
            /**
             * Static representation of the [[partnerFund]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_FUND: fieldBuilder.buildEdmTypeField('PartnerFund', 'Edm.String', true),
            /**
             * Static representation of the [[partnerGrant]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_GRANT: fieldBuilder.buildEdmTypeField('PartnerGrant', 'Edm.String', true),
            /**
             * Static representation of the [[budgetPeriod]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BUDGET_PERIOD: fieldBuilder.buildEdmTypeField('BudgetPeriod', 'Edm.String', true),
            /**
             * Static representation of the [[partnerBudgetPeriod]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PARTNER_BUDGET_PERIOD: fieldBuilder.buildEdmTypeField('PartnerBudgetPeriod', 'Edm.String', true),
            /**
             * Static representation of the [[branchCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BRANCH_CODE: fieldBuilder.buildEdmTypeField('BranchCode', 'Edm.String', true),
            /**
             * Static representation of the [[companyCodeCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            COMPANY_CODE_CURRENCY: fieldBuilder.buildEdmTypeField('CompanyCodeCurrency', 'Edm.String', true),
            /**
             * Static representation of the [[amountInCompanyCodeCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            AMOUNT_IN_COMPANY_CODE_CURRENCY: fieldBuilder.buildEdmTypeField('AmountInCompanyCodeCurrency', 'Edm.Decimal', true),
            /**
             * Static representation of the [[taxAmountInCoCodeCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_AMOUNT_IN_CO_CODE_CRCY: fieldBuilder.buildEdmTypeField('TaxAmountInCoCodeCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[taxBaseAmountInCoCodeCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_BASE_AMOUNT_IN_CO_CODE_CRCY: fieldBuilder.buildEdmTypeField('TaxBaseAmountInCoCodeCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[valuationDiffAmtInCoCodeCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            VALUATION_DIFF_AMT_IN_CO_CODE_CRCY: fieldBuilder.buildEdmTypeField('ValuationDiffAmtInCoCodeCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[cashDiscountAmtInCoCodeCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_AMT_IN_CO_CODE_CRCY: fieldBuilder.buildEdmTypeField('CashDiscountAmtInCoCodeCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[invoiceAmtInCoCodeCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_AMT_IN_CO_CODE_CRCY: fieldBuilder.buildEdmTypeField('InvoiceAmtInCoCodeCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[transactionCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TRANSACTION_CURRENCY: fieldBuilder.buildEdmTypeField('TransactionCurrency', 'Edm.String', true),
            /**
             * Static representation of the [[amountInTransactionCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            AMOUNT_IN_TRANSACTION_CURRENCY: fieldBuilder.buildEdmTypeField('AmountInTransactionCurrency', 'Edm.Decimal', true),
            /**
             * Static representation of the [[originalTaxBaseAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ORIGINAL_TAX_BASE_AMOUNT: fieldBuilder.buildEdmTypeField('OriginalTaxBaseAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[taxAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_AMOUNT: fieldBuilder.buildEdmTypeField('TaxAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[taxBaseAmountInTransCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_BASE_AMOUNT_IN_TRANS_CRCY: fieldBuilder.buildEdmTypeField('TaxBaseAmountInTransCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[withholdingTaxBaseAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            WITHHOLDING_TAX_BASE_AMOUNT: fieldBuilder.buildEdmTypeField('WithholdingTaxBaseAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[plannedAmtInTransactionCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PLANNED_AMT_IN_TRANSACTION_CRCY: fieldBuilder.buildEdmTypeField('PlannedAmtInTransactionCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[cashDiscountBaseAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_BASE_AMOUNT: fieldBuilder.buildEdmTypeField('CashDiscountBaseAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[cashDiscountAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CASH_DISCOUNT_AMOUNT: fieldBuilder.buildEdmTypeField('CashDiscountAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[netPaymentAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            NET_PAYMENT_AMOUNT: fieldBuilder.buildEdmTypeField('NetPaymentAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[withholdingTaxAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            WITHHOLDING_TAX_AMOUNT: fieldBuilder.buildEdmTypeField('WithholdingTaxAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[withholdingTaxExemptionAmt]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            WITHHOLDING_TAX_EXEMPTION_AMT: fieldBuilder.buildEdmTypeField('WithholdingTaxExemptionAmt', 'Edm.Decimal', true),
            /**
             * Static representation of the [[invoiceAmountInFrgnCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_AMOUNT_IN_FRGN_CURRENCY: fieldBuilder.buildEdmTypeField('InvoiceAmountInFrgnCurrency', 'Edm.Decimal', true),
            /**
             * Static representation of the [[balanceTransactionCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BALANCE_TRANSACTION_CURRENCY: fieldBuilder.buildEdmTypeField('BalanceTransactionCurrency', 'Edm.String', true),
            /**
             * Static representation of the [[amountInBalanceTransacCrcy]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            AMOUNT_IN_BALANCE_TRANSAC_CRCY: fieldBuilder.buildEdmTypeField('AmountInBalanceTransacCrcy', 'Edm.Decimal', true),
            /**
             * Static representation of the [[additionalCurrency1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ADDITIONAL_CURRENCY_1: fieldBuilder.buildEdmTypeField('AdditionalCurrency1', 'Edm.String', true),
            /**
             * Static representation of the [[valuationDiffAmtInAddlCrcy1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            VALUATION_DIFF_AMT_IN_ADDL_CRCY_1: fieldBuilder.buildEdmTypeField('ValuationDiffAmtInAddlCrcy1', 'Edm.Decimal', true),
            /**
             * Static representation of the [[amountInAdditionalCurrency1]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            AMOUNT_IN_ADDITIONAL_CURRENCY_1: fieldBuilder.buildEdmTypeField('AmountInAdditionalCurrency1', 'Edm.Decimal', true),
            /**
             * Static representation of the [[additionalCurrency2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ADDITIONAL_CURRENCY_2: fieldBuilder.buildEdmTypeField('AdditionalCurrency2', 'Edm.String', true),
            /**
             * Static representation of the [[amountInAdditionalCurrency2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            AMOUNT_IN_ADDITIONAL_CURRENCY_2: fieldBuilder.buildEdmTypeField('AmountInAdditionalCurrency2', 'Edm.Decimal', true),
            /**
             * Static representation of the [[valuationDiffAmtInAddlCrcy2]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            VALUATION_DIFF_AMT_IN_ADDL_CRCY_2: fieldBuilder.buildEdmTypeField('ValuationDiffAmtInAddlCrcy2', 'Edm.Decimal', true),
            /**
             * Static representation of the [[paymentCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PAYMENT_CURRENCY: fieldBuilder.buildEdmTypeField('PaymentCurrency', 'Edm.String', true),
            /**
             * Static representation of the [[amountInPaymentCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            AMOUNT_IN_PAYMENT_CURRENCY: fieldBuilder.buildEdmTypeField('AmountInPaymentCurrency', 'Edm.Decimal', true),
            /**
             * Static representation of the [[creditControlAreaCurrency]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CREDIT_CONTROL_AREA_CURRENCY: fieldBuilder.buildEdmTypeField('CreditControlAreaCurrency', 'Edm.String', true),
            /**
             * Static representation of the [[hedgedAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            HEDGED_AMOUNT: fieldBuilder.buildEdmTypeField('HedgedAmount', 'Edm.Decimal', true),
            /**
             * Static representation of the [[baseUnit]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BASE_UNIT: fieldBuilder.buildEdmTypeField('BaseUnit', 'Edm.String', true),
            /**
             * Static representation of the [[quantity]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            QUANTITY: fieldBuilder.buildEdmTypeField('Quantity', 'Edm.Decimal', true),
            /**
             * Static representation of the [[goodsMovementEntryUnit]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            GOODS_MOVEMENT_ENTRY_UNIT: fieldBuilder.buildEdmTypeField('GoodsMovementEntryUnit', 'Edm.String', true),
            /**
             * Static representation of the [[quantityInEntryUnit]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            QUANTITY_IN_ENTRY_UNIT: fieldBuilder.buildEdmTypeField('QuantityInEntryUnit', 'Edm.Decimal', true),
            /**
             * Static representation of the [[purchasingDocumentPriceUnit]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PURCHASING_DOCUMENT_PRICE_UNIT: fieldBuilder.buildEdmTypeField('PurchasingDocumentPriceUnit', 'Edm.String', true),
            /**
             * Static representation of the [[purchaseOrderQty]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PURCHASE_ORDER_QTY: fieldBuilder.buildEdmTypeField('PurchaseOrderQty', 'Edm.Decimal', true),
            /**
             * Static representation of the [[materialPriceUnitQty]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            MATERIAL_PRICE_UNIT_QTY: fieldBuilder.buildEdmTypeField('MaterialPriceUnitQty', 'Edm.Decimal', true),
            /**
             * Static representation of the [[numberOfItems]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            NUMBER_OF_ITEMS: fieldBuilder.buildEdmTypeField('NumberOfItems', 'Edm.Decimal', true),
            /**
             * Static representation of the [[accountingDocumentCreationDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_CREATION_DATE: fieldBuilder.buildEdmTypeField('AccountingDocumentCreationDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[creationTime]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            CREATION_TIME: fieldBuilder.buildEdmTypeField('CreationTime', 'Edm.Time', true),
            /**
             * Static representation of the [[lastChangeDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LAST_CHANGE_DATE: fieldBuilder.buildEdmTypeField('LastChangeDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[exchangeRateDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            EXCHANGE_RATE_DATE: fieldBuilder.buildEdmTypeField('ExchangeRateDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[accountingDocCreatedByUser]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOC_CREATED_BY_USER: fieldBuilder.buildEdmTypeField('AccountingDocCreatedByUser', 'Edm.String', true),
            /**
             * Static representation of the [[transactionCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TRANSACTION_CODE: fieldBuilder.buildEdmTypeField('TransactionCode', 'Edm.String', true),
            /**
             * Static representation of the [[intercompanyTransaction]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INTERCOMPANY_TRANSACTION: fieldBuilder.buildEdmTypeField('IntercompanyTransaction', 'Edm.String', true),
            /**
             * Static representation of the [[documentReferenceId]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            DOCUMENT_REFERENCE_ID: fieldBuilder.buildEdmTypeField('DocumentReferenceID', 'Edm.String', true),
            /**
             * Static representation of the [[recurringAccountingDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            RECURRING_ACCOUNTING_DOCUMENT: fieldBuilder.buildEdmTypeField('RecurringAccountingDocument', 'Edm.String', true),
            /**
             * Static representation of the [[reverseDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REVERSE_DOCUMENT: fieldBuilder.buildEdmTypeField('ReverseDocument', 'Edm.String', true),
            /**
             * Static representation of the [[reverseDocumentFiscalYear]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REVERSE_DOCUMENT_FISCAL_YEAR: fieldBuilder.buildEdmTypeField('ReverseDocumentFiscalYear', 'Edm.String', true),
            /**
             * Static representation of the [[accountingDocumentHeaderText]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_HEADER_TEXT: fieldBuilder.buildEdmTypeField('AccountingDocumentHeaderText', 'Edm.String', true),
            /**
             * Static representation of the [[exchangeRate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            EXCHANGE_RATE: fieldBuilder.buildEdmTypeField('ExchangeRate', 'Edm.Decimal', true),
            /**
             * Static representation of the [[businessTransactionType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BUSINESS_TRANSACTION_TYPE: fieldBuilder.buildEdmTypeField('BusinessTransactionType', 'Edm.String', true),
            /**
             * Static representation of the [[batchInputSession]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BATCH_INPUT_SESSION: fieldBuilder.buildEdmTypeField('BatchInputSession', 'Edm.String', true),
            /**
             * Static representation of the [[financialManagementArea]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            FINANCIAL_MANAGEMENT_AREA: fieldBuilder.buildEdmTypeField('FinancialManagementArea', 'Edm.String', true),
            /**
             * Static representation of the [[reversalIsPlanned]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REVERSAL_IS_PLANNED: fieldBuilder.buildEdmTypeField('ReversalIsPlanned', 'Edm.Boolean', true),
            /**
             * Static representation of the [[plannedReversalDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            PLANNED_REVERSAL_DATE: fieldBuilder.buildEdmTypeField('PlannedReversalDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[taxIsCalculatedAutomatically]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_IS_CALCULATED_AUTOMATICALLY: fieldBuilder.buildEdmTypeField('TaxIsCalculatedAutomatically', 'Edm.Boolean', true),
            /**
             * Static representation of the [[taxBaseAmountIsNetAmount]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_BASE_AMOUNT_IS_NET_AMOUNT: fieldBuilder.buildEdmTypeField('TaxBaseAmountIsNetAmount', 'Edm.Boolean', true),
            /**
             * Static representation of the [[sourceCompanyCode]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SOURCE_COMPANY_CODE: fieldBuilder.buildEdmTypeField('SourceCompanyCode', 'Edm.String', true),
            /**
             * Static representation of the [[logicalSystem]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LOGICAL_SYSTEM: fieldBuilder.buildEdmTypeField('LogicalSystem', 'Edm.String', true),
            /**
             * Static representation of the [[taxExchangeRate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_EXCHANGE_RATE: fieldBuilder.buildEdmTypeField('TaxExchangeRate', 'Edm.Decimal', true),
            /**
             * Static representation of the [[reversalReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REVERSAL_REASON: fieldBuilder.buildEdmTypeField('ReversalReason', 'Edm.String', true),
            /**
             * Static representation of the [[branch]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            BRANCH: fieldBuilder.buildEdmTypeField('Branch', 'Edm.String', true),
            /**
             * Static representation of the [[reference1InDocumentHeader]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_1_IN_DOCUMENT_HEADER: fieldBuilder.buildEdmTypeField('Reference1InDocumentHeader', 'Edm.String', true),
            /**
             * Static representation of the [[reference2InDocumentHeader]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            REFERENCE_2_IN_DOCUMENT_HEADER: fieldBuilder.buildEdmTypeField('Reference2InDocumentHeader', 'Edm.String', true),
            /**
             * Static representation of the [[invoiceReceiptDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            INVOICE_RECEIPT_DATE: fieldBuilder.buildEdmTypeField('InvoiceReceiptDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[ledger]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LEDGER: fieldBuilder.buildEdmTypeField('Ledger', 'Edm.String', true),
            /**
             * Static representation of the [[ledgerGroup]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LEDGER_GROUP: fieldBuilder.buildEdmTypeField('LedgerGroup', 'Edm.String', true),
            /**
             * Static representation of the [[alternativeReferenceDocument]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ALTERNATIVE_REFERENCE_DOCUMENT: fieldBuilder.buildEdmTypeField('AlternativeReferenceDocument', 'Edm.String', true),
            /**
             * Static representation of the [[taxReportingDate]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            TAX_REPORTING_DATE: fieldBuilder.buildEdmTypeField('TaxReportingDate', 'Edm.DateTime', true),
            /**
             * Static representation of the [[accountingDocumentClass]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            ACCOUNTING_DOCUMENT_CLASS: fieldBuilder.buildEdmTypeField('AccountingDocumentClass', 'Edm.String', true),
            /**
             * Static representation of the [[exchangeRateType]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            EXCHANGE_RATE_TYPE: fieldBuilder.buildEdmTypeField('ExchangeRateType', 'Edm.String', true),
            /**
             * Static representation of the [[latePaymentReason]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            LATE_PAYMENT_REASON: fieldBuilder.buildEdmTypeField('LatePaymentReason', 'Edm.String', true),
            /**
             * Static representation of the [[salesDocumentCondition]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            SALES_DOCUMENT_CONDITION: fieldBuilder.buildEdmTypeField('SalesDocumentCondition', 'Edm.String', true),
            /**
             * Static representation of the [[isReversal]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_REVERSAL: fieldBuilder.buildEdmTypeField('IsReversal', 'Edm.Boolean', true),
            /**
             * Static representation of the [[isReversed]] property for query construction.
             * Use to reference this property in query operations such as 'select' in the fluent request API.
             */
            IS_REVERSED: fieldBuilder.buildEdmTypeField('IsReversed', 'Edm.Boolean', true),
            ...this.navigationPropertyFields,
            /**
             *
             * All fields selector.
             */
            ALL_FIELDS: new odata_v2_1.AllFields('*', OperationalAcctgDocItemCube_1.OperationalAcctgDocItemCube)
        };
    }
}
exports.OperationalAcctgDocItemCubeApi = OperationalAcctgDocItemCubeApi;
