// @sap/cloud-sdk-vdm-accounting-document-service/service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountingDocumentService = exports.accountingDocumentService = void 0;
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This is a generated file powered by the SAP Cloud SDK for JavaScript.
 */
const OperationalAcctgDocItemCubeApi_1 = require("./OperationalAcctgDocItemCubeApi");
const odata_v2_1 = require("@sap-cloud-sdk/odata-v2");
const BatchRequest_1 = require("./BatchRequest");
function accountingDocumentService(deSerializers = odata_v2_1.defaultDeSerializers) {
    return new AccountingDocumentService((0, odata_v2_1.mergeDefaultDeSerializersWith)(deSerializers));
}
exports.accountingDocumentService = accountingDocumentService;
class AccountingDocumentService {
    constructor(deSerializers) {
        this.apis = {};
        this.deSerializers = deSerializers;
    }
    initApi(key, ctor) {
        if (!this.apis[key]) {
            this.apis[key] = new ctor(this.deSerializers);
        }
        return this.apis[key];
    }
    get operationalAcctgDocItemCubeApi() {
        return this.initApi('operationalAcctgDocItemCubeApi', OperationalAcctgDocItemCubeApi_1.OperationalAcctgDocItemCubeApi);
    }
    get batch() {
        return BatchRequest_1.batch;
    }
    get changeset() {
        return BatchRequest_1.changeset;
    }
}
exports.AccountingDocumentService = AccountingDocumentService;
